src/window.rs

```diff
@@ -439,8 +439,15 @@ pub fn ui_loop() {
                     ..
                 } => {
                     keycode = received_keycode;
+                    keytexts.push((keycode, None));
+                }
+                Event::TextInput { text, .. } => {
+                    if let Some(keycode) = keycode {
+                        if text.len() > 1 || keycode.name() != text.to_uppercase() {
```

Comment:
text.to_uppercase feels possibly like a hack to me... Can you explain what the goal is here? Maybe it would be better just to remove the check?

Reply:
Yeah, I think removing the check would be best also and I already have locally.

Reply:
I don't think SDL has an is_character function.